File: 349_Intersection/sol.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> num1(nums1.begin(), nums1.end());
        vector<int> i;
        for(std::vector<int>::iterator it = nums2.begin(); it != nums2.end(); ++it){
            if (num1.erase(*it)){
                i.push_back(*it);
            }
        }
        return i;
    }
};

File: 349_Intersection/sol.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        i = []
        num1 = set(nums1)
        for n in nums2:
            if n in num1 and n not in i:
                i.append(n)
        return i